utils.go

```diff
@@ -194,7 +198,9 @@ func (h *Headscale) getUsedIPs() (*netaddr.IPSet, error) {
 	// but this was quick to get running and it should be enough
 	// to begin experimenting with a dual stack tailnet.
 	var addressesSlices []string
-	h.db.Model(&Machine{}).Pluck("ip_addresses", &addressesSlices)
+	h.db.Model(&Machine{}).
```

Comment:
I dont think we safely can assume that we can hand out IPs that belong to expired nodes as of now, I think that requires a bit more thinking. 

I think this needs to be reverted, and a separate issue needs to be raised for discussion.

Reply:
I agree. This is probably not ideal. I added this as if we keep around expired records for the same machine then we will eventually run out of IPs. If we instead fix the handleAuthKey() to refresh the machine recorded if expired instead of inserting a new one this is not needed.